infrastructure/zk/src/docker.ts

```diff
@@ -31,40 +29,46 @@ async function dockerCommand(command: 'push' | 'build', image: string) {
         image = 'keybase-secret';
     }
     if (command == 'build') {
-        await _build(image);
+        await _build(image, tag);
     } else if (command == 'push') {
-        await _push(image);
+        await _push(image, tag);
     }
 }
 
-async function _build(image: string) {
+async function _build(image: string, tag: string) {
     if (image == 'nginx') {
         await utils.spawn('yarn explorer build');
     }
     if (image == 'server' || image == 'prover') {
         await contract.build();
     }
-    const { stdout: imageTag } = await utils.exec('git rev-parse --short HEAD');
-    const latestImage = `-t matterlabs/${image}:latest`;
-    const taggedImage = ['nginx', 'server', 'prover'].includes(image) ? `-t matterlabs/${image}:${imageTag}` : '';
-    await utils.spawn(`DOCKER_BUILDKIT=1 docker build ${latestImage} ${taggedImage} -f ./docker/${image}/Dockerfile .`);
+
+    const latestImage = `-t rsksmart/rollup-${image}:${tag}`;
+    await utils.spawn(`DOCKER_BUILDKIT=1 docker build ${latestImage} -f ./docker/${image}/Dockerfile .`);
 }
 
-async function _push(image: string) {
-    await utils.spawn(`docker push matterlabs/${image}:latest`);
-    if (['nginx', 'server', 'prover', 'event-listener'].includes(image)) {
-        const { stdout: imageTag } = await utils.exec('git rev-parse --short HEAD');
-        await utils.spawn(`docker push matterlabs/${image}:${imageTag}`);
-    }
+async function _push(image: string, tag: string) {
+    await utils.spawn(`docker push rsksmart/${image}:${tag}`);
+}
+
+export async function build(image: string, tag: string) {
+    await dockerCommand('build', image, tag);
+}
+
+export async function buildFromTag(tag: string) {
+    const splitted = tag.split(':');
+    await dockerCommand('build', splitted[0], splitted[1]);
```

Comment:
no such word, past tense of split is split ;)
also it doesn't describe the object enough. I'm guessing the tag is something like <image_name>:<tag>, right? Where is this pattern defined, or what could be taken as reference?
If my assumption is correct, then it would be clearer as something like this:
```suggestion
    const [image_name, tag] = tag.split(':');
    await dockerCommand('build', image_name, tag);
```

Reply:
updated